cpp/src/holtwinters/internal/hw_optim.cuh

```diff
@@ -871,7 +873,7 @@ void holtwinters_optim_gpu(const raft::handle_t& handle,
   int threads_per_block = 128;
 
   // How much sm needed for shared kernel
-  size_t sm_needed  = sizeof(Dtype) * threads_per_block * frequency;
+  int sm_needed     = sizeof(Dtype) * threads_per_block * frequency;
```

Comment:
It's not clear to me why `sm_needed` should be an `int` rather than a `size_t` here. I see that `threads_per_block` and `frequency` are both ints, so I understand where the warning is coming from, but do we want to instead either do some sort of safe casting to `std::size_t` or see if we can change the type for `frequency`?

Reply:
I don't think shared memory size needs to be larger than an int.

Reply:
I'm more worried about semantic intent here. `std::size_t` clearly indicates this is a "how much of something?" quantity whereas `int` communicates something much less specific. That being said, given our discussion about the need for follow-up to this PR, I'd be fine leaving it for that.